iceoryx_dds/include/iceoryx_dds/internal/gateway/iox_to_dds.inl

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_DDS_INTERNAL_GATEWAY_IOX_TO_DDS_INL
+#define IOX_DDS_INTERNAL_GATEWAY_IOX_TO_DDS_INL
+
+#include "iceoryx_dds/internal/log/logging.hpp"
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+
+#include <chrono>
+#include <thread>
+
+#include "iceoryx_dds/gateway/iox_to_dds.hpp"
+
+namespace iox
+{
+namespace dds
+{
+// ======================================== Public ======================================== //
+template <typename channel_t, typename gateway_t>
+inline Iceoryx2DDSGateway<channel_t, gateway_t>::Iceoryx2DDSGateway() noexcept
+    : gateway_t()
+{
+}
+
+template <typename channel_t, typename gateway_t>
+inline void Iceoryx2DDSGateway<channel_t, gateway_t>::loadConfiguration(const GatewayConfig& config) noexcept
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] Configuring gateway.";
+    for (const auto& service : config.m_configuredServices)
+    {
+        if (!this->findChannel(service).has_value())
+        {
+            setupChannel(service);
+        }
+    }
+}
+
+
+template <typename channel_t, typename gateway_t>
+inline void Iceoryx2DDSGateway<channel_t, gateway_t>::discover(const iox::capro::CaproMessage& msg) noexcept
+{
+    iox::LogDebug() << "[Iceoryx2DDSGateway] <CaproMessage> "
+                    << iox::capro::CaproMessageTypeString[static_cast<uint8_t>(msg.m_type)]
+                    << " { Service: " << msg.m_serviceDescription.getServiceIDString()
+                    << ", Instance: " << msg.m_serviceDescription.getInstanceIDString()
+                    << ", Event: " << msg.m_serviceDescription.getEventIDString() << " }";
+
+    if (msg.m_serviceDescription.getServiceIDString() == iox::capro::IdString("Introspection"))
```

Comment:
This is something you could change with the next PR.
Add a `constexpr const char INTROSPECTION_SERVICE_ID[] = "Introspection";` to introspection_types.hpp and use that here. This makes the code more robust

Reply:
It has been done. Coming soon :)